sql/sql_class.cc

```diff
@@ -6197,6 +6197,11 @@ int THD::decide_logging_format(TABLE_LIST *tables)
   DBUG_RETURN(0);
 }
 
+sql_mode_t sql_mode_for_dates(THD *thd)
+{
+	return thd->variables.sql_mode &
+		(MODE_NO_ZERO_DATE | MODE_NO_ZERO_IN_DATE | MODE_INVALID_DATES);
+}
```

Comment:
Because we should calculate the now() or current_timestamp() from THD, and store it in the dictionary.